src/debugger/debugtool.test.js

```diff
@@ -1,55 +1,113 @@
 import Chronicle from "./chronicle"
 import DebugTool from "./debugtool"
+import DebugView from "./debugview"
 
-describe("Debug Tool, when intercepting keyValue calls,", () => {
-  beforeEach(() => {
-    jest.spyOn(global, "Date").mockImplementation(() => ({
-        getTime: () => 1234,
-      }))
-    Chronicle.init()
+jest.mock("./debugview")
+
+describe("Debug Tool", () => {
+  beforeAll(() => {
+    jest.useFakeTimers({ now: 1234 })
   })
 
-  afterEach(() => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+
     DebugTool.tearDown()
     Chronicle.tearDown()
+
+    Chronicle.init()
   })
 
-  it("should always add entry to chronicle if the key does not match one of the defined static keys", () => {
-    const testObj1 = { key: "bitrate", value: "1000" }
-    const testObj2 = { key: "imNotSpecial", value: "nobodylovesme" }
-    const testObj3 = { key: "idontmatch", value: "pleaseaddme" }
+  describe("logging an error", () => {
+    it("takes a string", () => {
+      DebugTool.error("something went wrong")
+
+      expect(Chronicle.retrieve()).toEqual([
+        { type: "error", error: new Error("something went wrong"), timestamp: 1234 },
+      ])
+
+      DebugTool.toggleVisibility()
+
+      expect(DebugView.render).toHaveBeenCalledWith(
+        expect.objectContaining({
+          dynamic: [expect.stringContaining("Error: something went wrong")],
+        })
+      )
+    })
 
-    const expectedArray = [
-      { type: "keyvalue", keyvalue: testObj1, timestamp: 1234 },
-      { type: "keyvalue", keyvalue: testObj2, timestamp: 1234 },
-      { type: "keyvalue", keyvalue: testObj3, timestamp: 1234 },
-    ]
+    it("takes an Error", () => {
+      DebugTool.error(new TypeError("something went REALLY wrong"))
 
-    DebugTool.keyValue(testObj1)
-    DebugTool.keyValue(testObj2)
-    DebugTool.keyValue(testObj3)
+      expect(Chronicle.retrieve()).toEqual([
+        { type: "error", error: new TypeError("something went REALLY wrong"), timestamp: 1234 },
+      ])
 
-    const chronicle = Chronicle.retrieve()
+      DebugTool.toggleVisibility()
 
-    expect(chronicle).toEqual(expectedArray)
+      expect(DebugView.render).toHaveBeenCalledWith(
+        expect.objectContaining({
+          dynamic: [expect.stringContaining("TypeError: something went REALLY wrong")],
+        })
+      )
+    })
   })
 
-  it("overwrites a keyvalue entry to the chronicle if that keyvalue already exists", () => {
-    const testObj = { key: "akey", value: "something" }
-    const testObj1 = { key: "bitrate", value: "1000" }
-    const testObj2 = { key: "bitrate", value: "1001" }
+  describe("logging a warning", () => {
+    it("takes a string", () => {
+      DebugTool.warn("you're using a deprecated thingie!")
```

Comment:
Maybe we should const this string too so we don't need to update it in multiple locations again, the test on L67 can interpolate it in?

Reply:
Same as previous comment